src/main/java/org/brackit/xquery/function/fn/EndsWith.java

```diff
@@ -0,0 +1,45 @@
+package org.brackit.xquery.function.fn;
+
+import org.brackit.xquery.QueryContext;
+import org.brackit.xquery.atomic.Int32;
+import org.brackit.xquery.atomic.QNm;
+import org.brackit.xquery.function.AbstractFunction;
+import org.brackit.xquery.module.StaticContext;
+import org.brackit.xquery.xdm.Item;
+import org.brackit.xquery.xdm.Sequence;
+import org.brackit.xquery.xdm.Signature;
+
+public class EndsWith extends AbstractFunction {
+    public EndsWith(QNm name, Signature signature) {
+        super(name, signature, true);
+    }
+
+    public boolean isZeroLength(Sequence[] args) {
+        if (args[0] == null || ((Item) args[0]).atomize().stringValue().length() == 0 && args[1] == null
+                || ((Item) args[1]).atomize().stringValue().length() > 0) {
+            return false;
+        } else if (args[1] == null || ((Item) args[1]).atomize().stringValue().length() == 0 && args[0] == null
+                || ((Item) args[0]).atomize().stringValue().length() > 0) {
+            return true;
+        } else {
+            return false;
+        }
+
+    }
+
+    @Override
+    public Sequence execute(StaticContext sctx, QueryContext ctx, Sequence[] args) {
+        if (isZeroLength(args) == Boolean.TRUE) {
+            return Int32.ONE;
+        }
+        if (isZeroLength(args) == Boolean.FALSE) {
+            return Int32.ZERO;
+        }
+        String first = ((Item) args[0]).atomize().stringValue();
+        String second = ((Item) args[1]).atomize().stringValue();
+        boolean result = first.contains(second);
```

Comment:
Seems to be strange, that you compare the strings with contains, but the function is called `fn:ends-with`. Second, it should return a Boolean value. Third, the function has to be registered and test cases would be nice :-)

Reply:
Yeah looking at it again using `contains()` was silly but that should be fixed in my newest commit. In regards to your second point about Boolean value... my initial implementation returned `Boolean.TRUE` and `Boolean.FALSE` but there is a type mismatch with the expected return type Sequence so I switched to Int32

Reply:
Regarding your last point: I will add a test case as soon as I can

Reply:
You need the type `Bool` which implements `Sequence`. Here are examples you can use for your tests:

https://www.w3.org/TR/xpath-functions-31/#func-ends-with